### DISCUSSION TITLE
Empty input array creates single closing tag

### DISCUSSION THREAD

**Discussion by NebsterSK** [spatie/array-to-xml#221] [Q&A]
Hello everyone, is it possible to turn off creation of single closing tag when input array is empty?

This code:

`ArrayToXml::convert(['ticket' => []], 'tickets', xmlEncoding: 'UTF-8');`

creates this output:

`<?xml version="1.0" encoding="UTF-8"?>
<tickets>
    <ticket/>
</tickets>`

however I need this output:

`<?xml version="1.0" encoding="UTF-8"?>
<tickets>
</tickets>`

Is that possible?

I went trough the docs but couldnt find any option / property.

**Comment by dimitri-koenig** [comment]
@NebsterSK, unfortunately that won't work. And one could argue that is more of a feature than a bug. BUT you can surround it with `array_filter` which would make it work as desired: `array_filter(['ticket' => []])`
